starter-code/js/index.js

```diff
@@ -1,52 +1,85 @@
-function deleteItem(e){
+var calculatePrices = document.getElementsByClassName('btn-success')[0];
 
-}
-
-function getPriceByProduct(itemNode){
-
-}
-
-function updatePriceByProduct(productPrice, index){
-
-}
-
-function getTotalPrice() {
-
-}
-
-function createQuantityInput(){
-
-}
+// Puts all item prices into an array
 
-function createDeleteButton(){
+// Calculate Button produces total for each item line and grand total
 
-}
-
-function createQuantityNode(){
+calculatePrices.onclick = function() {
 
+var grandTotal = 0;
+var itemPrice = [];
+for (i=0; i<document.getElementsByClassName('price').length; i++){
+  itemPrice.push(parseFloat(document.getElementsByClassName('price')[i].innerHTML.trim().slice(1)));
 }
+var priceArray = itemPrice;
+  for (i=0; i<document.getElementsByClassName('price').length; i++){
+    var totalPrice = document.getElementsByClassName('totalprice')[i];
+    console.log(totalPrice);
+    var input = document.getElementsByClassName('quantity')[i].value;
+    console.log(input);
+    var total = parseInt(input) * priceArray[i];
+    console.log(total);
+    grandTotal += total;
+    totalPrice.innerHTML = "$" + total;
+  }
+  document.getElementById('grandtotal').innerHTML = "$" + grandTotal;
+};
 
-function createItemNode(dataType, itemData){
-
-}
-
-function createNewItemRow(itemName, itemUnitPrice){
-
-}
+// delete buttons
+var container = document.querySelector('.container');
 
-function createNewItem(){
+  for (i=0; i<document.getElementsByClassName("itemline").length;i++) {
+    var deleteButtons = document.getElementsByClassName('btn-delete')[i];
+    var itemLine = document.getElementsByClassName('itemline')[i];
+    deleteButtons.onclick = function (ev) {
+    var x = ev.currentTarget;
+    console.log(x);
+    console.log(x.parentNode.parentNode);
+    container.removeChild(x.parentNode.parentNode);
+    };
 
 }
-
-window.onload = function(){
-  var calculatePriceButton = document.getElementById('calc-prices-button');
-  var createItemButton = document.getElementById('new-item-create');
-  var deleteButtons = document.getElementsByClassName('btn-delete');
-
-  calculatePriceButton.onclick = getTotalPrice;
-  createItemButton.onclick = createNewItem;
-
-  for(var i = 0; i<deleteButtons.length ; i++){
-    deleteButtons[i].onclick = deleteItem;
-  }
+// add new items ////////////////////
```

Comment:
This is a tried and true method of creating new elements and adding them to the DOM but another method you might find easier is to use string interpolation and insert a whole string rather than create variables for everything.

So instead of doing what you did below, you can try 

var newItemHtml = `<div class="itemline"><div class="item">**"+ itemName +"**</div><div class="price">**"+ price +"**</div><div><label for="quantity">QTY</label><input name="quantity" class ="quantity" type="text" value="0" /></div><div class="totalprice">$0.00</div><div><button class="btn-delete">Delete</button></div></div>`

$('.container').appendChild(newItemHtml);


This method is a little more complex because it really easy to mess the html up when you write it in one line but a good way to not mess up is just to double click the part that you want to make dynamic with a variable and once it is highlighted, press delete. After that you insert a " + variableName + " in it place.

If you need a little bit more of an explanation on this method feel free to just ask me or Nizar and we will explain it to you in a little bit of more detail.